synapse/storage/databases/main/events.py

```diff
@@ -2081,7 +2081,9 @@ def _handle_redact_relations(
             txn.execute(sql, (redacted_relates_to, RelationTypes.THREAD))
 
             row = txn.fetchone()
-            # If a new latest event is found, update the threads table.
+            # If a latest event is found, update the threads table, this might
+            # be the same current latest event (if an earlier event in the thread
+            # was redacted).
```

Comment:
I was going to ask on an earlier commit: could this be a no-op update? Answer: yes. But that's fine.

Reply:
I couldn't figure out a way to make this "nicer". We could also pull the latest thread info from the `threads` table first and if it matches then no-op, but that doesn't seem to be any "better" than just upserting?

Reply:
Yeah, unless you already have the relevant info in memory I think this is completely fine.